packages/swing-store/src/transcriptStore.js

```diff
@@ -193,44 +225,101 @@ export function makeTranscriptStore(db, ensureTxn, noteExport) {
    */
   function rolloverSpan(vatID) {
     ensureTxn();
-    const { hash, startPos, endPos, size } = getCurrentSpanBounds(vatID);
-    const rec = spanRec(vatID, startPos, endPos, hash, size);
+    const { hash, startPos, endPos } = getCurrentSpanBounds(vatID);
+    const rec = spanRec(vatID, startPos, endPos, hash, 0);
     noteExport(historicSpanMetadataKey(rec), JSON.stringify(rec));
     sqlEndCurrentSpan.run(vatID);
-    sqlWriteSpan.run(vatID, endPos, endPos, initialHash, 1, 0);
+    sqlWriteSpan.run(vatID, endPos, endPos, initialHash, 1);
+    const newRec = spanRec(vatID, endPos, endPos, initialHash, 1);
+    noteExport(currentSpanMetadataKey(newRec), JSON.stringify(newRec));
+    if (!keepTranscripts) {
+      sqlDeleteOldItems.run(vatID, endPos);
+    }
+  }
+
+  const sqlDeleteVatSpans = db.prepare(`
+    DELETE FROM transcriptSpans
+    WHERE vatID = ?
+  `);
+
+  const sqlDeleteVatItems = db.prepare(`
+    DELETE FROM transcriptItems
+    WHERE vatID = ?
+  `);
+
+  const sqlGetVatSpans = db.prepare(`
+    SELECT startPos
+    FROM transcriptSpans
+    WHERE vatID = ?
+    ORDER BY startPos
+  `);
+  sqlGetVatSpans.pluck(true);
+
+  /**
+   * Delete all transcript data for a given vat (for use when, e.g., a vat is terminated)
+   *
+   * @param {string} vatID
+   */
+  function deleteVatTranscripts(vatID) {
+    ensureTxn();
+    const deletions = sqlGetVatSpans.all(vatID);
+    for (const startPos of deletions) {
+      const exportRec = spanRec(vatID, startPos);
+      noteExport(historicSpanMetadataKey(exportRec), undefined);
```

Comment:
I don't understand.  This will generate one `noteExport` per record for each record that is actually there.  Where do you see the extra one coming from?  Am I missing something?  I don't see any weird stuff showing up in the tests.

Reply:
Unless I'm mistaken, we would have previously exported `transcript.vXX.0`, `transcript.vXX.100` and `transcript.vXX.current`. The loop here will try to delete `transcript.vXX.0`, `transcript.vXX.100` and `transcript.vXX.200`, followed by the explicit `transcript.vXX.current`. The delete of `transcript.vXX.200` is extraneous.

Reply:
OK (per the discussion below), I know see what you're getting at.  With transcripts, the current span export data and the historic span export data are disjoint sets, and should always be treated as such.   I'm not sure the same logic applies to snapshots (your other comment immediately above) but I will look into that next.